### PR TITLE
Add SLES 15 and AL2023 support, remove binary build supporting files

### PR DESCRIPTION
**Pull request by gregbdunn**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

This adds support for detecting SLES 15 as well as the associated unit test, as well as removing the supporting files for binary builds, as support for binary builds was removed in a prior commit.

### CODE REVIEW

**Comment by gregbdunn** [comment]
With the release of AL2023 I've also added support for it with this PR, including updating modules and related tests.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,14 +2,16 @@
...
-* [Enhancement] Add support for RHEL 8.x and Ubuntu 22.10
+* [Enhancement] Add support for RHEL 8.x, Ubuntu 22.10, SLES 15, Amazon Linux 2023
 * [Enhancement] Add support for Python 3.7 to 3.10+
 
 #### Framework
 
 #### Modules
 * [Misc] Create Ubuntu specific copies of bcc modules to reflect the Debian package adding -bpfcc to the binary file names.
 * [New Module] Add enadiag net diagnostic module.
+* [New Module] Add sosreport os gather module.
```
Comment (Drudenhaus) [inline CHANGELOG.md:13]: Belongs in pull [91](https://github.com/awslabs/aws-ec2rescue-linux/pull/91) else this is dependent upon it.

**Review by Drudenhaus — COMMENTED** [review]
Are all these modules (or at least the programs they use) tested to work on AL2023 and SLES15?

**Comment by gregbdunn** [comment]
> Are all these modules (or at least the programs they use) tested to work on AL2023 and SLES15?

There are working versions available for them, though some, like the messages module, will not work if there has not been rsyslog added after the fact. The in module error checking should handle these situations, though, and I don't want to disable the module in case someone has made the choice to install them.

With no rsyslog shipping by default in al2023 we do need to revisit the journal module as well as some diagnostic modules that make use of the messages files. I was planning on handling those as a separate commit, as it is possible to setup syslog for compatibility with these and I did not want to necessarily outright disable them from being used on al2023. We likely need some additional discussion there on how we want to handle the deprecation of syslog-style log files among the larger distros.

File: CHANGELOG.md (hunk 1 above)
Comment (gregbdunn) [reply CHANGELOG.md]: Fixed

**Comment by gregbdunn** [comment]
New version of journal module added with safer defaults than the prior disabled version.

Still todo is handle the 7 diag modules that currently rely on searching messages/syslog files

**Comment by gregbdunn** [comment]
I have split out the journalctl search from logsearch and created a new logsearcxh_with_journal function (hat tip to obetsr there) and updated the 6 diagnostic modules that make use of it to now support journal searching.

This should bring us everything necessary for al2023 as well as proper journal support for any other distributions that move to journald.

Modern versions of journald do have built in grep support but it is a relatively new change and the journald versions I found in the defaults for these distros is too old to support it, so output is just piped into grep.

File: mod.d/hungtasks.yaml [hunk 2]
```diff
@@ -20,6 +20,9 @@ version: !!str 1.0
 title: !!str Detects hung tasks
 helptext: !!str |
   Detects hung tasks and gathers output (generally due to io device unresponsiveness)
+  For systems using journald instead of traditional syslog, the following optional flags are available.
+  Optional --since= shows entries on or newer than a specified date. Takes systemd.time format. Defaults to 48hr.
```
Comment (Drudenhaus) [inline mod.d/hungtasks.yaml:24]: Need to update the constraints. Applies to most modules updated in this commit (except softlockup).

**Review by Drudenhaus — CHANGES_REQUESTED** [review]

File: mod.d/hungtasks.yaml (hunk 2 above)
Comment (gregbdunn) [reply mod.d/hungtasks.yaml:24]: Done, good catch!

**Comment by gregbdunn** [comment]
Drudenhaus is currently unable to 'officially' review due to some permissions issues, but he's given the OK here so merging this.

**Review by Drudenhaus — APPROVED** [review]
LGTM now. Definitely needs squash merge per slack convo.

**Pull request merged**